diagnostics/seaice_suite/seaice_suite_sic_mean_sigma.py

```diff
@@ -0,0 +1,415 @@
+        #dask='parallelized')
+
+    rlag.name = 'lagcorr'
+    rlag=rlag.transpose('month',...)
+        
+    return rlag
+
+
+# In[8]:
+
+
+def processandsave(field,file_out,firstyr=1979,lastyr=2014):
+    ### 2) Doing computations on model or obs: #####################################
+    # this takes about a few min on data that is already limited to the Arctic
+    # recommend using DASK if you decide to compute global metrics
+    # residuals are detrended field for each month, effectively making them detrended and deseasonalized
+    themean, thestd, trend, detrendedstd, residuals = mainmonthlystats(field,firstyr,lastyr)
+    print('main stats done')
+    onemolagcorr=lagcorr(residuals,1) 
+    oneyrlagcorr=lagcorr(residuals,12)
+    onemolagcorr.name='onemolagcorr'
+    oneyrlagcorr.name='oneyrlagcorr'
+
+    ### 3) Saving output data: #####################################################
+
+    allstats=xr.merge([themean, thestd, trend, detrendedstd, onemolagcorr, oneyrlagcorr])
+
+    months = ['Jan','Feb','Mar','Apr','May','Jun','Jul','Aug','Sep','Oct','Nov','Dec']
+
+    allstats.coords['monthabbrev'] = xr.DataArray(months, dims='month')
+    allstats
+
+    allstats.to_netcdf(file_out)
+
+
+# In[11]:
+
+
+if processmod:
+    processandsave(field,proc_mod_file,firstyr,lastyr)
+
+if processobs:
+    processandsave(obs,proc_obs_file,firstyr,lastyr)
+
+
+# In[12]:
+
+
+### 4) Read processed data, regrid model to obs grid, plot, saving figures: #######################################
+
+obsstats = xr.open_dataset(proc_obs_file)
+obsstats=obsstats.rename({'latitude':'lat'})
+obsstats=obsstats.rename({'longitude':'lon'})
+
+obsmean=obsstats.themean 
+obsstd=obsstats.thestd
+obstrend=obsstats.trend 
+obsdetrendedstd=obsstats.detrended_std 
+obsonemolagcorr=obsstats.onemolagcorr
+obsoneyrlagcorr=obsstats.oneyrlagcorr
+
+modstats = xr.open_dataset(proc_mod_file)
+
+coords = [a for a in modstats.coords]
+if 'longitude' in coords:
+    modstats=modstats.rename({'latitude':'lat'})
+    modstats=modstats.rename({'longitude':'lon'})
+
+# regrid model data to obs grid
+method = 'nearest_s2d'       #method = 'nearest_d2s'  # this was bad do not use
+regridder = xe.Regridder(modstats, obsstats, method, periodic=False, reuse_weights=False)
+modstats=regridder(modstats)
+
+modmean=modstats.themean 
+modstd=modstats.thestd
+modtrend=modstats.trend 
+moddetrendedstd=modstats.detrended_std 
+modonemolagcorr=modstats.onemolagcorr
+modoneyrlagcorr=modstats.oneyrlagcorr
+
+month=modstats.month.values
+monthabbrev=modstats.monthabbrev.values
+
+obsstats.close()
+modstats.close()
+
+
+# In[13]:
+
+
+#import copy
+
+def monthlyplot(field, obs=None, edgec=None, figfile='./figure.png',
+                cmapname='PuBu_r', statname='Mean', unitname='Fraction',vmin=0.,vmax=1.):
+    fig = plt.figure(figsize=(12,10))
+    cmap_c = cmapname
+    #cmap = plt.get_cmap(cmapname)
+    #cmap_c = copy.copy(cmap)
+    #cmap_c.set_bad(color = 'lightgrey')
```

Comment:
Remove the commented lines; otherwise added an explanation for why they are retained.